workshops/util.py

```diff
@@ -137,47 +137,29 @@ def verify_upload_person_task(data):
         email = item.get('email', None)
         personal = item.get('personal', None)
         family = item.get('family', None)
+        person_id = item.get('existing_person_id', None)
         person = None
 
-        if email:
+        if person_id:
             try:
-                # check if first and last name matches person in the database
-                person = Person.objects.get(email__iexact=email)
-
-                for which, actual, uploaded in (
-                        ('personal', person.personal, personal),
-                        ('family', person.family, family)
-                ):
-                    if (actual == uploaded) or (not actual and not uploaded):
-                        pass
-                    else:
-                        errors.append('{0} mismatch: database "{1}" '
-                                      'vs uploaded "{2}".'
-                                      .format(which, actual, uploaded))
-
-            except Person.DoesNotExist:
-                # in this case we need to add a new person
-                pass
-
+                person = Person.objects.get(id=int(person_id))
+            except (ValueError, TypeError, Person.DoesNotExist):
```

Comment:
You validate `existing_person_id` inside `person_bulk_add_match_person` to make sure it's an `int`. So you can use `id=person_id` and maybe add an `assert isinstance(person_id, int)`.